src/peer.rs

```diff
@@ -399,9 +399,36 @@ impl PeerManager {
         self.send_channels[&peer.conn_info].send(msg);
     }
 
-    pub async fn flush_pipeline(&mut self) {
-        // wait for every peer's pipeline to become empty, as well as queue
-        // easiest way is to just use mpsc and have each peer communicate when no work left
+    pub async fn flush_pipeline(&mut self, peer: &mut RemotePeer) {
```

Comment:
This is good for flushing at a peer level, we also need a construct to "join" on all peer flushing at a top level. Like the comments said:
- Wait for every peer's pipeline to become empty, as well as queue
- Easiest way is to just use mpsc and have each peer communicate when no work left